src/dirtyfields/dirtyfields.py

```diff
@@ -164,3 +172,9 @@ def reset_state(sender, instance, **kwargs):
         instance._original_state = new_state
     if instance.ENABLE_M2M_CHECK:
         instance._original_m2m_state = instance._as_dict_m2m()
+    if instance.RESET_STATE_ON_SAVE_ONLY:
+        instance._m2m_dirty = False
```

Comment:
Should this be more like:

```
    if instance.RESET_STATE_ON_SAVE_ONLY:
        instance._m2m_dirty = False
    elif instance.ENABLE_M2M_CHECK:
         instance._original_m2m_state = instance._as_dict_m2m()
```

to avoid unnecessarily storing the original_m2m_state since we don't do anything with it in RESET_STATE_ON_SAVE_ONLY mode?

Reply:
yeah that's true